src/main/webapp/assets/js/productpage.js

```diff
@@ -0,0 +1,94 @@
+async function getMockData() {
+  let response = await fetch('/mockdata');
+  let mockdata = await response.json();
+  return mockdata;
+}
+
+async function populateProductDetails() {  
+  business = await getMockData();
+
+  populateBusinessDescription(business);
+  populateImageGallery(business.photoBlobstoreUrlList);
+  populateBusinessWriteup(business);
+  populateContactDetails(business);
+}
+
+function populateBusinessDescription(business) {
+  Array.from(document.getElementsByClassName('businessName')).forEach(element => element.innerHTML += business.name);
+
+  document.getElementById("businessLogo").src = business.logoBlobstoreUrl;
+
+  const categoryContainer = document.getElementById("businessCategories");
+  business.categories.forEach(category => categoryContainer.appendChild(createCategoryElement(category)));
+
+  document.getElementById("businessUrl").href = business.websiteUrl;
+
+  const ratingContainer = document.getElementById("aggregateRating");
+  //TODO: Fix star CSS and use CSS to set float rating
+}
+
+function populateImageGallery(photoUrlList) {
+  const imagePreviewGallery = document.getElementById("image-thumbnail-gallery");
+  const modalGalleryContainer = document.getElementById("slides-container");
+  const modalGalleryImagePreviewContainer = document.getElementById("gallery-thumbnail-container");
+
+  var image;
+  const urlListLength = photoUrlList.length;
+  for (image = 1; image <= urlListLength; image++) {
+    const url = photoUrlList[image-1];
+    imagePreviewGallery.innerHTML += createThumbnailImageElement(url);
+    modalGalleryContainer.innerHTML += createGalleryImageElement(url, image, urlListLength);
+    modalGalleryImagePreviewContainer.innerHTML += createGalleryImagePreviewElement(url, image);
+  }  
+}
+
+function populateBusinessWriteup(business) {
+  document.getElementById("businessDescr").innerHTML = business.description;
+  document.getElementById("menuUrl").href = business.menuUrl;
+}
+
+function populateContactDetails() {
+  document.getElementById("orderInfo").innerHTML = business.orderInformation;
+  document.getElementById("contactUrl").href = business.contactUrl;
```

Comment:
Take note that I've changed contactUrl to contactInfo in my PR so when you merge in something might not work here.

I renamed it because the method of contact may be through other things other than a URL - could be a phont number, an instagram page, facebook page, website etc, so maybe we should leave it open-ended.

Reply:
hmm, in this case then we can't hyperlink it automatically right? Because based on our mock example, the link redirects to the Insta page, but if it's a phone number we shouldn't make a hyperlink

Reply:
No we won't hyper link it. It will be a paragraph of sorts, just like the order details. That's oky! Just a slight change. Merge this in soon?